File: java/infrastructure/src/main/java/dev/ufadevconf/infrastructure/repository/InMemoryPickupPointRepository.java
package dev.ufadevconf.infrastructure.repository;

import dev.ufadevconf.domain.PickupPoint;
import dev.ufadevconf.domain.PickupPointRepository;
import org.springframework.stereotype.Repository;

import java.util.Map;
import java.util.Optional;
import java.util.concurrent.ConcurrentHashMap;

@Repository
public class InMemoryPickupPointRepository implements PickupPointRepository {

    private final Map<String, PickupPoint> storage = new ConcurrentHashMap<>();

    @Override
    public void save(PickupPoint pickupPoint) {
        storage.put(pickupPoint.getCode(), pickupPoint);
    }

    @Override
    public Optional<PickupPoint> findByCode(String code) {
        return Optional.ofNullable(storage.get(code));
    }
}

File: kotlin/infrastructure/src/main/kotlin/dev/ufadevconf/infrastructure/repository/InMemoryPickupPointRepository.kt
package dev.ufadevconf.infrastructure.repository

import dev.ufadevconf.domain.PickupPoint
import dev.ufadevconf.domain.PickupPointRepository
import org.springframework.stereotype.Repository
import java.util.concurrent.ConcurrentHashMap

@Repository
class InMemoryPickupPointRepository : PickupPointRepository {

    private val storage = ConcurrentHashMap<String, PickupPoint>()

    override fun save(pickupPoint: PickupPoint) {
        storage[pickupPoint.code] = pickupPoint
    }

    override fun findByCode(code: String): PickupPoint? = storage[code]
}
